### ISSUE TITLE
sort by country that is related to resource not directly but through the third relation is not work properly

### ISSUE THREAD

**Issue by Ivanov-Anton** [JSONAPI-Resources/jsonapi-resources#1409]
## This issue is a (choose one):

- [x] Problem/bug report.
- [ ] Feature request.
- [ ] Request for support. **Note: Please try to avoid submitting issues for support requests. Use [Gitter](https://gitter.im/cerebris/jsonapi-resources) instead.** 

## Checklist before submitting:

- [x] I've searched for an existing issue.
- [ ] I've asked my question on [Gitter](https://gitter.im/cerebris/jsonapi-resources) and have not received a satisfactory answer.
- [ ] I've included a complete [bug report template](https://github.com/cerebris/jsonapi-resources/blob/master/lib/bug_report_templates/rails_5_master.rb). This step helps us and allows us to see the bug without trying to reproduce the problem from your description. It helps you because you will frequently detect if it's a problem specific to your project.
- [ ] The feature I'm asking for is compliant with the [JSON:API](http://jsonapi.org/) spec.

## Description

When making a request to a specific API endpoint with the query parameter sort=country.name, the application crashes due to an error in the sorting mechanism. The problem arises because the country relationship does not directly belong to the destination model but rather through another association. This leads to an undefined column error when attempting to perform the sorting operation.

## Additional info

```
class Routing::Destination < ApplicationRecord
  belongs_to :network_prefix, class_name: 'System::NetworkPrefix', optional: true
  has_one :country, through: :network_prefix
end

class Api::Rest::Admin::Routing::DestinationResource < JSONAPI::Resource
  has_one :country, class_name: 'Country', force_routed: true, foreign_key_on: :related
end

class System::NetworkPrefix < ApplicationRecord
  belongs_to :country, class_name: 'System::Country', foreign_key: :country_id, optional: true
end

```

```
request that lead to 500 error
GET /destinations?sort=country.name
```

```
SELECT "class4"."destinations".* FROM "class4"."destinations"
LEFT JOIN sys.countries AS country_sorting ON country_sorting.id = class4.destinations.country_id
LEFT JOIN class4.rate_groups AS rate_group_sorting ON rate_group_sorting.id = class4.destinations.rate_group_id
ORDER BY country_sorting.name asc, rate_group_sorting.name asc LIMIT $1 OFFSET $2  [["LIMIT", 50], ["OFFSET", 0]]


Internal Server Error: PG::UndefinedColumn: ERROR:  column destinations.country_id does not exist
LINE 1: ...ntries AS country_sorting ON country_sorting.id = class4.des...
                                                             ^

```

jsonapi-resource version 0.9.12

**Comment by johnnymo87** [comment]
In the 0.9 version of the gem, the sorting on relationships **_[only works for `belongs_to` relationships](https://github.com/cerebris/jsonapi-resources/blob/v0.9.12/lib/jsonapi/resource.rb#L738)_**. I believe this is the actual bug here.

**Comment by Ivanov-Anton** [comment]
@johnnymo87 Yeah correct. I'll try to improve this method to fix this in release-0-9 branch

**Comment by baranyeni** [comment]
As a workaround I plucked the ids from the query results and generated another query like this `Model.where(id: ids)` by so, sorting is possible since there are no relations etc. and it works even I don't like how it performs
